cmake/FindPythonModule.cmake

```diff
@@ -0,0 +1,35 @@
+# Downloaded from
+#   http://www.cmake.org/pipermail/cmake/2011-January/041666.html
+# * Added CMake stage dir to PYTHONPATH
+# * Added FORCE to location var
+# * Function to macro so module_FOUND shows up
+# * Remove ``if(NOT PY_${module})`` so runs each time, also remove module caps
+# * Module name, not path in fphsa
+
+#.rst:
+#
+# Find if a Python module is installed.
+# Usage: find_python_module(PyQt4 REQUIRED)
+
+macro(find_python_module module)
+    unset(PY_${module} CACHE)
+    if(ARGC GREATER 1 AND ARGV1 STREQUAL "REQUIRED")
+        set(${module}_FIND_REQUIRED TRUE)
+    endif()
+    # A module's location is usually a directory, but for binary modules
+    # it's a .so file.
+    execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c" 
+                            "import re, sys; \
+                             sys.path.insert(0, '${STAGED_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}${PYMOD_INSTALL_LIBDIR}'); \
+                             import ${module}; \
+                             print(re.compile('/__init__.py.*').sub('', ${module}.__file__))"
```

Comment:
Cool! Shouldn't this be `sys.path.insert(1,...)`. Granted this could be an off case due to how this is being run, but worth pondering + a small note why it is `0`.

Reply:
Normally, yes, `insert(1`. I was having such problems with it detecting the clone on my system that I threw the hammer at it with `0`. 

The whole detecting python mods setup is flimsy and w/o many of the tweaking tools we have for `projectConfig.cmake` projects. We may need to substantially develop the machinery. For instance, there's legitimate reason a user would want his own location in `sys.path[0]`. But then, even if we delicately tuned detection, we can't do the same to runtime (don't want to set the user's `PYTHONPATH`), just as `core.so`'s `RPATH` is crude compared to all the `_DIR` cmake vars.

So, long story short, I don't know what's best here in the long term. But for now, I'll clobber with `0` and add a note. :-)